replicator/src/main.rs

```
@@ -61,12 +64,32 @@ async fn main_impl() -> Result<(), Box<dyn Error>> {
         publication,
     } = settings.source;
 
+    let TlsSettings {
+        trusted_root_certs,
+        enabled,
+    } = settings.tls;
+
+    let mut trusted_root_certs_vec = vec![];
+    let ssl_mode = if enabled {
```

we either have `on` or `off`, I think a third option is missing. If `enabled` but no trusted_root_certs is supplied, we should use `require` for the TlsMode.

sure, not verifying the certificate is similar to running without encryption, but it still offers a degree of protection and ensures passive network sniffing can't read transmitted data.

---

I've tried to keep it absolutely simple. We can add the `require` mode which as you said gives us transport encryption but no MITM protection, but I'm not sure why would anyone want to do that. Is that because the user for some reason couldn't obtain the root cert? For Supabase platform we plan to always enable TLS, so for us it works as is.

Postgres's TLS options are needlessly complex IMO and actually hurt TLS adoption. I'd argue forcing people to either completely disable TLS or pick the strictest option will nudge them to pick properly configured TLS.

One thing I can do though is to fail fast with a proper error message when TLS is enabled but no root cert is provided. Although even this won't catch all misconfigurations early, like when a wrong cert is provided.

But, if I'm wrong and users do need the `require` mode we can always add it later. In other words, keeping this as is is a type 2 decision which we can flip later. What do you think?

---

> Postgres's TLS options are needlessly complex IMO and actually hurt TLS adoption.

Agree with you 100%.

> Is that because the user for some reason couldn't obtain the root cert
Yes, basically this. 

> One thing I can do though is to fail fast with a proper error message when TLS is enabled but no root cert is provided.

This would improve the DX, and avoids some of the complexity that you mentioned 👍🏼